### PR TITLE
1469 data and code pathway

### LINKED ISSUE #1469: Customized forms for the Data & Code Pathway
Fields for this pathway are listed below. In our current set up the creators are added in their own form and the embargo date occurs during the work settings. It may be useful to consider moving the embargo setting somewhere in the deposit form.  
Work Title (Required)
Work Type (Required)
Description (Required)
Publication Date (Required)
Subtitle (Optional)
Keyword(s) (Optional)
Publisher (Required) -> should this be set to ScholarSphere automatically?
Related URLs (Optional)
Geographic Area (Optional)
Funding Reference (Optional)
Subject (Optional)
Language (Optional)
Source (Optional)
Semantic Version (Optional)
Creator (Required)
Depositor (Required)
Embargo Date (Optional)
Version (Optional)
License (Required)

Additional features for this pathway:
- Curation request when work is in draft status prior to publication. This would result in the following options for the depositor:
1. Review and then publish (recommended): Curators will review your submission before it is published. A draft DOI can be requested before data is reviewed. The dataset is not published until curation is complete and the submitter is ready for the files to be public.   Include some sort of recognition (badge/message) that the work went through curatorial review. 
2. Publish then review: Data/code is self-submitted to the repository and is immediately public. A working DOI can be requested after you complete your submission. Note, because curatorial review occurs after publication, any changes or updates may result in a new version.  

- Disable Penn State Only Access

### PR DESCRIPTION
**Pull request by EricDurante**
resolves https://github.com/psu-libraries/scholarsphere/issues/1469

### CODE REVIEW

File: spec/features/dashboard/work_version_form_spec.rb [hunk 1]
```diff
@@ -243,6 +243,122 @@
         end
       end
     end
+
+    context 'when selecting a work type that uses the data and code deposit pathway' do
+      it 'shows only the fields for scholarly works' do
```
Comment (ajkiessl) [inline spec/features/dashboard/work_version_form_spec.rb]: This should be fields for data and code.

**Review by ajkiessl — CHANGES_REQUESTED** [review]
This all looks good to me.  There was only one thing I saw and it's very minor.

File: spec/features/dashboard/work_version_form_spec.rb (hunk 1 above)
Comment (EricDurante) [reply spec/features/dashboard/work_version_form_spec.rb]: Good eye!

**Review by ajkiessl — DISMISSED** [review]
lgtm

**Comment by EricDurante** [comment]
Sorry, @ajkiessl - I had to rebase this again and resolve a conflict, so I need another review

**Review by ajkiessl — APPROVED** [review]

**Pull request merged**